### ISSUE TITLE
docker logs since timestamp not accurate

### ISSUE THREAD

**Issue by devisions** [docker/cli#2552]
**Description**

`docker logs` does not consider the `--since {timestamp}` option.

**Steps to reproduce the issue:**
1. Started a container.
2. Stopped the container.
3. Showing the log entries, hopefully only the ones logged since the shutdown (using the classic option: `docker stop fsd_blog_db`), but it shows older the entries (of the entire day, maybe?) also. See the output below as a concrete example.

**Describe the results you received:**

```shell
$ docker logs --since 2020-05-27T20:24:00 fsd_blog_db

PostgreSQL Database directory appears to contain a database; Skipping initialization

2020-05-27 19:58:54.711 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2020-05-27 19:58:54.711 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2020-05-27 19:58:54.713 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2020-05-27 19:58:54.776 UTC [28] LOG:  database system was interrupted; last known up at 2020-05-25 18:48:31 UTC
2020-05-27 19:58:58.370 UTC [28] LOG:  database system was not properly shut down; automatic recovery in progress
2020-05-27 19:58:58.426 UTC [28] LOG:  redo starts at 0/16516E0
2020-05-27 19:58:58.426 UTC [28] LOG:  invalid record length at 0/16517C0: wanted 24, got 0
2020-05-27 19:58:58.426 UTC [28] LOG:  redo done at 0/1651788
2020-05-27 19:58:58.567 UTC [1] LOG:  database system is ready to accept connections
2020-05-27 20:24:22.143 UTC [1] LOG:  received smart shutdown request
2020-05-27 20:24:22.149 UTC [1] LOG:  background worker "logical replication launcher" (PID 34) exited with exit code 1
2020-05-27 20:24:22.149 UTC [29] LOG:  shutting down
2020-05-27 20:24:22.185 UTC [1] LOG:  database system is shut down
$ 
```

**Describe the results you expected:**

The expectation is to show only the log entries since the provided timestamp, according to the `--since` option.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b
 Built:             Wed Mar 11 01:21:11 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       afacb8b
  Built:            Wed Mar 11 01:29:16 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Output of `docker info`:**

```
Client:
 Debug Mode: false

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 1
 Server Version: 19.03.8
 Storage Driver: overlay2
  Backing Filesystem: <unknown>
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 7
 Total Memory: 3.355GiB
 Name: docker-desktop
 ID: IGQP:E7QS:NUPH:WDM2:I37B:K3EZ:N3RP:Q5DM:R62G:7T3E:R64I:4I26
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: gateway.docker.internal:3128
 HTTPS Proxy: gateway.docker.internal:3129
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Running on a standard macOS Catalina, version 10.15.4.

**Comment by thaJeztah** [comment]
Does it work correctly if you use `--since 2020-05-27T20:24:00Z` (UTC)?

**Comment by devisions** [comment]
@thaJeztah Good point!
With `Z` suffix (aka UTC / no timezone) works correctly.
So I got mislead by the `docker logs -h` that gives an example without `Z`:
```
...
  -f, --follow         Follow log output
      --since string   Show logs since timestamp (e.g. 2013-01-02T13:23:37) or relative (e.g. 42m for 42 minutes)
...
```

**Comment by devisions** [comment]
And this applies to `--until` option also.
I noticed that a description of the whole behavior exists in [docs/reference/commandline/service_logs.md](https://github.com/devisions/cli/blob/master/docs/reference/commandline/service_logs.md).

As a complete (explicit) example, I would include the `Z` in the help example also.
If that's ok, I could send a small PR. This is example is present in a couple of places.

**Comment by devisions** [comment]
@thaJeztah Thanks for the feedback, Sebastiaan!
Yes, I've noticed that a couple of places would need to be updated for consistency reasons.

I made PR #2555 . I hope it's ok and really glad that I could contribute. 😊 

I would consider putting that putting an explicit time zone or just UTC helps since in may cases the logs on that container has a timestamp, therefore it would be easier to reason and refer to them when providing that timestamp instead of considering your local system's time zone.
